packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -110,26 +107,16 @@ export class Accordion extends GOVUKFrontendComponent {
   $showAllText = null
 
   /**
-   * @param {Element | null} $root - HTML element to use for accordion
+   * @param {HTMLElement | null} $root - HTML element to use for accordion
```

Comment:
**issue** I think this will cause issue with the `createAll` function or more generally `document.querySelectorAll` as the later [returns `Element`](https://www.typescriptlang.org/docs/handbook/dom-manipulation.html#the-queryselector-and-queryselectorall-methods) (unless the selector is a tag name, but we use `[data-module="..."]` in `createAll`). 

Could the child class constructors be kept as `Element`?